internal/pkg/runtime/engine/apptainer/container_linux.go

```diff
@@ -792,24 +792,29 @@ func (c *container) mountImage(mnt *mount.Point) error {
 
 	mountType := mnt.Type
 
-	if mountType == "encryptfs" {
+	if mountType == "encryptfs" || mountType == "gocryptfs" {
 		key, err = mount.GetKey(mnt.InternalOptions)
 		if err != nil {
 			return err
 		}
 	}
 
-	if imageDriver != nil && imageDriver.Features()&image.ImageFeature != 0 {
-		params := &image.MountParams{
-			Source:     mnt.Source,
-			Target:     mnt.Destination,
-			Filesystem: mountType,
-			Flags:      flags,
-			Offset:     offset,
-			Size:       sizelimit,
-			Key:        key,
-			FSOptions:  opts,
-		}
+	params := &image.MountParams{
+		Source:     mnt.Source,
+		Target:     mnt.Destination,
+		Filesystem: mountType,
+		Flags:      flags,
+		Offset:     offset,
+		Size:       sizelimit,
+		Key:        key,
+		FSOptions:  opts,
+	}
+
+	if mountType == "gocryptfs" && imageDriver != nil && imageDriver.GetDriverName() == "gocryptfsDriver" {
+		return imageDriver.Mount(params, c.rpcOps.Mount)
+	}
+
+	if imageDriver != nil && imageDriver.Features()&image.ImageFeature != 0 && imageDriver.GetDriverName() == "fuseapps" {
```

Comment:
I don't understand how imageDriver can have both gocryptfsDriver and fuseapps as the name at the same time.  I thought they would be all one driver.

Reply:
addressed in these commits https://github.com/apptainer/apptainer/pull/1185/commits/ea57976b196239860c32afb4de903f27a0d3051f https://github.com/apptainer/apptainer/pull/1185/commits/c315e9fb461150fbc00b0aa3f94e50569dd90317